### ISSUE TITLE
Global Styles Variations

### ISSUE THREAD

**Issue by shazahm1** [WordPress/gutenberg#39323]
### Description

The "custom" settings in the theme.json are not overridden by the global styles variations syles:

- https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-json/#custom

### Step-by-step reproduction instructions

1. [Add a Custom CSS Property to the theme.json file](https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-json/#custom).
2. [Create a style variation preset](https://developer.wordpress.org/block-editor/how-to-guides/themes/create-block-theme/#global-styles-presets).
3. Add the same Custom CSS property, but set with a different value.
4. Choose the style variation in the Site Editor.
5. Inspect the page source.
6. The Custom CSS Property value is that from the theme.json file and not the style variation as expected.

### Environment info

- WordPress 5.9.1
- Gutenberg 12.7.1
- Theme Check 20211203
- Theme Sniffer 1.1.2 
- PHP 7.4.28

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by paaljoachim** [comment]
@oandregal Here is another issue you might want to know about.

**Comment by ndiego** [comment]
While this issue needs to be addressed for completeness, @oandregal do you think this is something we can reasonably fix by 6.0 with Beta 3 scheduled for tomorrow? If not, I will remove it from the project board. Thanks!

**Comment by oandregal** [comment]
Hey all, I've given this a test. This is what I did:

- Use TwentyTwentyTwo.
- Created a style variation under `styles/custom.json` that only modifies the values of the `custom` section in use by TwentyTwentyTwo:

```json
{
  "version": 2,
  "title": "Custom",
  "settings": {
    "custom": {
      "typography": {
        "fontSize": {
          "huge": "12px",
          "gigantic": "13px",
          "colossal": "14px"
        },
        "lineHeight": {
          "tiny": 2.2,
          "small": 2.4,
          "medium": 2.8,
          "normal": 3.2
        }
      }
    }
  }
}
```

- Go to the style variation list and pick the "Custom" one. This is what I see:

https://user-images.githubusercontent.com/583546/165260459-16e0805f-b1a1-4412-af8d-23dcad3afea3.mp4

- If I publish the post and go to the front-end I also see the new values.

As far as my testing goes, this is working as expected. @shazahm1 would you share an use case in which is not working for you? @paaljoachim @ndiego are you able to reproduce?

**Comment by shazahm1** [comment]
I've attached both the `theme.json` and the style variation `dark.json` for you to use and hopefully replicate.

In the `theme.json` file, I have colors set in the `settings > custom > color` section:

```
		"custom": {
			"color": {
				"-brand": "var(--pink-6)",
				"-surface-1": "var(--gray-0)",
				"-surface-2": "var(--gray-2)",
				"-surface-3": "var(--gray-3)",
				"-surface-4": "var(--gray-4)"
			}
		},
```

In the `dark.json` style variation, I have the same custom properties but with different values:

```
		"custom": {
			"color": {
				"-brand": "var(--pink-4)",
				"-surface-1": "var(--gray-9)",
				"-surface-2": "var(--gray-8)",
				"-surface-3": "var(--gray-7)",
				"-surface-4": "var(--gray-6)"
			}
		}
```

When the `dark.json` style variation is used and I inspect the page source I find `--wp--custom--color--surface-4: var(--gray-4)` instead of the expected `--wp--custom--color--surface-4: var(--gray-6)`.

I hope this helps!

[theme.zip](https://github.com/WordPress/gutenberg/files/8563106/theme.zip)

**Comment by shazahm1** [comment]
As a quick follow-up...

I guess this is not an issue after all. I added `"title": "Dark",` to the style variation, and it worked correctly. I noticed that in your working example, so I thought I would try adding it. So, I guess the `title` key has significance as to whether the custom properties are used in the style variation. 

I have `$schema` set `https://schemas.wp.org/trunk/theme.json`, and it is reported as an invalid property key by the IDE. I suppose this will straighten itself out when WP6.0 is released. Hopefully, it'll be highlighted as a required property.

**Comment by ndiego** [comment]
Great find @shazahm1! Thank you for following up, was just about to test. Do you feel comfortable closing this out? I can look into adding a separate issue for the schema update if it's not already taken care of in 6.0.

**Comment by shazahm1** [comment]
Yeah, I'll close. It turns out this is not related to the `title` prop at all. It seems choosing the "Default" variation and saving in the Site Editor and then choosing the style variation and saving is what resolved the issue. Perhaps something changed since I opened the issue that fixed this because that is an action I took many times while working with the style variations. Since I hit that roadblock, I stopped working on my block theme and did not look into it again until today.

Thanks for your time!
